dev/tools/lib/git.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+/// A wrapper around git process calls that can be mocked for unit testing.
+class Git {
+  Git(this.processManager) : assert(processManager != null);
+
+  final ProcessManager processManager;
+
+  String getOutput(
+    List<String> args,
+    String explanation, {
+    @required String workingDirectory,
+  }) {
+    final ProcessResult result = _run(args, workingDirectory);
+    if ((result.stderr as String).isEmpty && result.exitCode == 0)
+      return (result.stdout as String).trim();
+    _reportFailureAndExit(args, workingDirectory, result, explanation);
```

Comment:
Just a note, once this is opted into null safety you can give `_reportFailureAndExit` a return type of `Never` and then there will be no analyzer warning